lib/segment/src/data_types/index.rs

```diff
@@ -198,6 +202,129 @@ pub struct TextIndexParams {
     // todo(phrase_match): remove skip
     #[serde(skip)]
     pub phrase_matching: Option<bool>,
+
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub stopwords: Option<StopwordsInterface>,
+}
+
+#[derive(Debug, Serialize, Deserialize, JsonSchema, Clone, PartialEq, Hash, Eq)]
+#[serde(untagged)]
+pub enum StopwordsInterface {
+    Language(Language),
+    Set(StopwordsSet),
+}
+
+impl StopwordsInterface {
+    #[cfg(feature = "testing")]
+    pub fn new_custom(custom: &[&str]) -> Self {
+        StopwordsInterface::Set(StopwordsSet {
+            languages: None,
+            custom: Some(custom.iter().map(|s| (*s).to_string()).collect()),
+        })
+    }
+
+    #[cfg(feature = "testing")]
+    pub fn new_language(language: Language) -> Self {
+        StopwordsInterface::Language(language)
+    }
+
+    #[cfg(feature = "testing")]
+    pub fn new_set(languages: &[Language], custom: &[&str]) -> Self {
+        StopwordsInterface::Set(StopwordsSet {
+            languages: Some(languages.iter().cloned().collect()),
+            custom: Some(custom.iter().map(|s| (*s).to_string()).collect()),
+        })
+    }
+}
+
+#[derive(
+    Debug, Serialize, Deserialize, JsonSchema, Clone, PartialEq, PartialOrd, Ord, Hash, Eq,
+)]
+#[serde(rename_all = "snake_case")]
+pub enum Language {
+    #[serde(alias = "ar")]
+    Arabic,
+    #[serde(alias = "az")]
+    Azerbaijani,
+    #[serde(alias = "eu")]
+    Basque,
+    #[serde(alias = "bn")]
+    Bengali,
+    #[serde(alias = "ca")]
+    Catalan,
+    #[serde(alias = "zh")]
+    Chinese,
+    #[serde(alias = "da")]
+    Danish,
+    #[serde(alias = "nl")]
+    Dutch,
+    #[serde(alias = "en")]
+    English,
+    #[serde(alias = "fi")]
+    Finnish,
+    #[serde(alias = "fr")]
+    French,
+    #[serde(alias = "de")]
+    German,
+    #[serde(alias = "el")]
+    Greek,
+    #[serde(alias = "he")]
+    Hebrew,
+    #[serde(alias = "hi-en")]
+    Hinglish,
+    #[serde(alias = "hu")]
+    Hungarian,
+    #[serde(alias = "id")]
+    Indonesian,
+    #[serde(alias = "it")]
+    Italian,
+    #[serde(alias = "kk")]
+    Kazakh,
+    #[serde(alias = "ne")]
+    Nepali,
+    #[serde(alias = "no")]
+    Norwegian,
+    #[serde(alias = "pt")]
+    Portuguese,
+    #[serde(alias = "ro")]
+    Romanian,
+    #[serde(alias = "ru")]
+    Russian,
+    #[serde(alias = "sl")]
+    Slovene,
+    #[serde(alias = "es")]
+    Spanish,
+    #[serde(alias = "sv")]
+    Swedish,
+    #[serde(alias = "tg")]
+    Tajik,
+    #[serde(alias = "tr")]
+    Turkish,
+}
+impl fmt::Display for Language {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        let json_string = serde_json::to_string(self).map_err(|_| fmt::Error)?;
+        f.write_str(json_string.trim_matches('"'))
```

Comment:
Maybe it is a bit cleaner this way?
```suggestion
    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
        write!(f, "{self:?}")
```